popup/popup.js

```diff
@@ -1638,6 +1768,35 @@ function waitForAccount() {
       .catch(errorHandler(waitForAccount));
 }
 
+function waitForAllAccounts() {
+    showAlert('#alert_accounts');
+    global.alert = '#alert_accounts';
+    let heading = document.querySelector('#alert_accounts .alert_heading');
+    heading.textContent = 'Loading accounts...';
+
+    let discovered = [];
+    let discovering = null;
+
+    let onStart = (account) => {
+        discovering = account;
+    };
+
+    let onUsed = () => {};
+
+    let onEnd = () => {
+        discovered.push(discovering);
+        discovering = null;
+    };
+
+    return new Promise((resolve, reject) => {
```

Comment:
Also I am not sure what this promise does... it returns just the first account? Why do you want that?